python/lsst/daf/butler/registry/queries/_builder.py

```diff
@@ -94,50 +101,13 @@ def joinDimensionElement(self, element: DimensionElement):
             associated with a database table (see `DimensionElement.hasTable`).
         """
         assert element not in self._elements, "Element already included in query."
-        assert element.hasTable(), "Cannot join element with no table."
-        table = self._dimensionStorage[element].getElementTable(self.summary.dataId)
-        if element in self.summary.spatial:
-            self.joinToCommonSkyPix(element)
-            self._columns.regions[element] = table.columns[REGION_FIELD_SPEC.name]
-        joinDimensions = list(element.graph.required)
-        joinDimensions.extend(element.implied)
-        joinOn = self.startJoin(table, joinDimensions, element.RecordClass.__slots__)
-        if element in self.summary.temporal:
-            intervalInTable = Timespan(
-                begin=table.columns[TIMESPAN_FIELD_SPECS.begin.name],
-                end=table.columns[TIMESPAN_FIELD_SPECS.end.name],
-            )
-            for intervalInQuery in self._columns.timespans.values():
-                joinOn.append(intervalInQuery.overlaps(intervalInTable, ops=sqlalchemy.sql))
-            self._columns.timespans[element] = intervalInTable
-        self.finishJoin(table, joinOn)
-        self._elements[element] = table
-
-    def joinToCommonSkyPix(self, element: DimensionElement):
-        """Add the table relating a spatial `DimensionElement` to the
-        universe's `~DimensionUniverse.commonSkyPix` dimension to the query.
-
-        External calls to this method should rarely be necessary; `finish` will
-        automatically call it if the `DimensionElement` has been identified as
-        one that must be included.
-
-        Parameters
-        ----------
-        element : `DimensionElement`
-            Element for which the relationship should be added.  The element
-            must be associated with a database table (see
-            `DimensionElement.hasTable`) and must have
-            `Dimensionelement.spatial` `True`.
-        """
-        if element is self.summary.universe.commonSkyPix:
-            return
-        assert element in self.summary.spatial,\
-            f"{element}'s spatial information is not relevant for this query."
-        table = self._dimensionStorage[element].getCommonSkyPixOverlapTable(self.summary.dataId)
-        assert table is not None, f"No relationship found from the common skypix to {element}."
-        dimensions = NamedValueSet(element.graph.required)
-        dimensions.add(self.summary.universe.commonSkyPix)
-        self.joinTable(table, dimensions)
+        storage = self._dimensionStorage[element]
```

Comment:
do you want a normal lookup failure or something with a custom message if the element is not in the dimensionStorage

Reply:
Normal failure is fine for now; I think this is guaranteed to succeed with the only `DimensionRecordStorageManager` implementation we have right now, and how to handle manager implementations that do not guarantee this (because they're in a multi-layer Registry in which only the base layer has tables for all dimensions) is sufficiently unsettled that I'm sure this line will have to change in other ways if we don't continue to guarantee its success.